src/lightning_app/core/app.py

```diff
@@ -520,6 +526,84 @@ def state_dict(self) -> Dict:
     def load_state_dict(self, state: Dict) -> None:
         self.set_state(state)
 
+    @classmethod
+    def _load_checkpoint_from_json_file(cls, path: str) -> None:
+        checkpoint_dict = json.load(open(path))
+        # TODO: Checkpoint versioning and compatibility checks
+        return checkpoint_dict
+
+    def _get_checkpoint_if_available_locally(self, checkpoint: str) -> Optional[dict]:
+        checkpoint_dict = None
+        if os.path.exists(checkpoint):
+            checkpoint_dict = self._load_checkpoint_from_json_file(checkpoint)
+        if os.path.exists(os.path.join(self.checkpoint_dir, checkpoint)):
+            checkpoint_dict = self._load_checkpoint_from_json_file(os.path.join(self.checkpoint_dir, checkpoint))
+
+        return checkpoint_dict
+
+    def _get_checkpoint_if_available_on_drive(self, checkpoint: str) -> Optional[dict]:
```

Comment:
Can this and perhaps the one above become utility functions? We could construct the path here in the app and then call the utility function with the already constructed path.

Reply:
Why moving it to the utility function? I am not sure if it should be reused outside of the app class.